src/main.rs

```diff
@@ -1,41 +1,52 @@
-mod actions;
+mod interface;
 mod models;
 
 mod store {
-    use crate::models;
+    use crate::models::Ship;
 
     pub struct Session {
-        pub shipyard: Vec<models::Ship>,
-        current_ship: usize,
+        pub shipyard: Vec<Ship>,
+        index: usize // current ship index, implements a circular buffer
```

Comment:
I wouldn't recommend using only `index` for this variable, it's too vague, that's why you had to add the comment. Could we use a better, self-describing name, e.g. `ship_index`?